### PR TITLE
fix: mssql commentTemplate handling table object

### LINKED ISSUE #14078: Mssql - Comment on column throws errors @model.sync
## Issue Creation Checklist

- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

The following line does not work:
https://github.com/sequelize/sequelize/blob/f69d76108d15d5ee4a94ae5f425faac665f8edce/src/dialects/mssql/query-generator.js#L276

`this.quoteIdentifier` should be `this.quoteTable`

This only occurs when a comment is set on a table column, when running model.sync.

### What do you expect to happen?

A newly created table with the set comments.

### What is actually happening?

The following error is thrown:

```
UnhandledPromiseRejectionWarning: TypeError: identifier.replace is not a function
    at MSSQLQueryGenerator.quoteIdentifier [as _quoteIdentifier] (/home/DOWNUNDER/jos/Workspace/service_scenario/build/server/node_modules/sequelize/lib/dialects/mssql/query-generator.js:798:27)
    at MSSQLQueryGenerator.quoteIdentifier (/home/DOWNUNDER/jos/Workspace/service_scenario/build/server/node_modules/sequelize/lib/dialects/abstract/query-generator.js:645:19)
    at MSSQLQueryGenerator.commentTemplate (/home/DOWNUNDER/jos/Workspace/service_scenario/build/server/node_modules/sequelize/lib/dialects/mssql/query-generator.js:255:194)
    at MSSQLQueryGenerator.createTableQuery (/home/DOWNUNDER/jos/Workspace/service_scenario/build/server/node_modules/sequelize/lib/dialects/mssql/query-generator.js:136:30)
    at MSSqlQueryInterface.createTable (/home/DOWNUNDER/jos/Workspace/service_scenario/build/server/node_modules/sequelize/lib/dialects/abstract/query-interface.js:93:31)
    at Function.sync (/home/DOWNUNDER/jos/Workspace/service_scenario/build/server/node_modules/sequelize/lib/model.js:930:5)
```

### Environment

- Sequelize version: 6.16.1
- Node.js version: 14.16.1

### How does this problem relate to dialects?

- [ ] I think this problem happens regardless of the dialect.
- [x] I think this problem happens only for the following dialect(s): mssql (/ tedious)
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start. (see 'bug description')
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### PR DESCRIPTION
**Pull request by KreativJos**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [x] Have you added new tests to prevent regressions? (does not apply - edited existing test)
- [x] Does `yarn test` or `yarn test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)? (does not apply)
- [x] Did you update the typescript typings accordingly (if applicable)? (does not apply)
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Fixes #14078 - Issue & unit test

### CODE REVIEW

**Comment by WikiRik** [comment]
Some tests are unfortunately flaky, no worries about that. We'll review and ignore those failing tests for now

File: test/unit/dialects/mssql/query-generator.test.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ if (current.dialect.name === 'mssql') {
...
+      expectsql(this.queryGenerator.createTableQuery({ tableName: 'myTable' }, { int: 'INTEGER COMMENT Foo Bar', varchar: 'VARCHAR(50) UNIQUE COMMENT Bar Foo' }, {}), { mssql: 'IF OBJECT_ID(\'[myTable]\', \'U\') IS NULL CREATE TABLE [myTable] ([int] INTEGER, [varchar] VARCHAR(50) UNIQUE); EXEC sp_addextendedproperty @name = N\'MS_Description\', @value = N\'Foo Bar\', @level0type = N\'Schema\', @level0name = \'dbo\', @level1type = N\'Table\', @level1name = [myTable], @level2type = N\'Column\', @level2name = [int]; EXEC sp_addextendedproperty @name = N\'MS_Description\', @value = N\'Bar Foo\', @level0type = N\'Schema\', @level0name = \'dbo\', @level1type = N\'Table\', @level1name = [myTable], @level2type = N\'Column\', @level2name = [varchar];' });
```
Comment (ephys) [inline test/unit/dialects/mssql/query-generator.test.js:96]: From what I can see, `createTableQuery` accepts `TableName | string`. The test should be compatible with just a string version too.

`quoteTable` is supposed to be compatible with `TableName | string`. If you're encountering a crash, could you fix `quoteTable`? (its typings are also wrong in query-interface.d.ts)

**Review by ephys — CHANGES_REQUESTED** [review]
Finally had time to review this

PR looks good, just one comment

File: test/unit/dialects/mssql/query-generator.test.js (hunk 1 above)
Comment (KreativJos) [reply test/unit/dialects/mssql/query-generator.test.js:96]: The issue is not with `quoteTable`, but with `quoteIdentifier` being called in the current mssql query-generator.
I will create duplicate the test, so both options will be tested.

Comment (ephys) [reply test/unit/dialects/mssql/query-generator.test.js:100]: For slightly more context, the reason I posted this comment was that I was surprised `'myTable'` was replaced with `{ tableName: 'myTable' }` ; and wondered why

**Review by ephys — APPROVED** [review]
Assuming the CI gives us greenlight, this looks good to me

**Comment by WikiRik** [comment]
@KreativJos Thanks for your work on this! We'll let you know when this is released on v6

**Pull request merged**